gensim/test/test_nmf.py

```diff
@@ -0,0 +1,291 @@
+from gensim import matutils, utils
+from gensim.test import basetmtests
+from gensim.test.utils import datapath, get_tmpfile, common_texts
+
+dictionary = Dictionary(common_texts)
+corpus = [dictionary.doc2bow(text) for text in common_texts]
+
+
+def testRandomState():
+    testcases = [np.random.seed(0), None, np.random.RandomState(0), 0]
+    for testcase in testcases:
+        assert (isinstance(utils.get_random_state(testcase), np.random.RandomState))
+
+
+class TestNmf(unittest.TestCase, basetmtests.TestBaseTopicModel):
+    def setUp(self):
+        self.corpus = mmcorpus.MmCorpus(datapath('testcorpus.mm'))
+        self.class_ = nmf.Nmf
+        self.model = self.class_(corpus, id2word=dictionary, num_topics=2, passes=100)
+
+    def testTransform(self):
+        # create the transformation model
+        model = self.class_(id2word=dictionary, num_topics=2, passes=100)
+        model.update(self.corpus)
+
+        # transform one document
+        doc = list(corpus)[0]
+        transformed = model[doc]
+
+        vec = matutils.sparse2full(transformed, 2)  # convert to dense vector, for easier equality tests
+        expected = [0., 1.]
+        # must contain the same values, up to re-ordering
+        self.assertTrue(np.allclose(sorted(vec), sorted(expected), atol=1e-1))
+
+    @unittest.skip('top topics is not implemented')
+    def testTopTopics(self):
+        top_topics = self.model.top_topics(self.corpus)
+
+        for topic, score in top_topics:
+            self.assertTrue(isinstance(topic, list))
+            self.assertTrue(isinstance(score, float))
+
+            for v, k in topic:
+                self.assertTrue(isinstance(k, six.string_types))
+                self.assertTrue(np.issubdtype(v, float))
+
+    def testGetTopicTerms(self):
+        topic_terms = self.model.get_topic_terms(1)
+
+        for k, v in topic_terms:
+            self.assertTrue(isinstance(k, numbers.Integral))
+            self.assertTrue(np.issubdtype(v, float))
+
+    def testGetDocumentTopics(self):
+
+        model = self.class_(
+            self.corpus, id2word=dictionary, num_topics=2, passes=100, random_state=np.random.seed(0)
+        )
+
+        doc_topics = model.get_document_topics(self.corpus)
+
+        for topic in doc_topics:
+            self.assertTrue(isinstance(topic, list))
+            for k, v in topic:
+                self.assertTrue(isinstance(k, numbers.Integral))
+                self.assertTrue(np.issubdtype(v, float))
+
+        # Test case to use the get_document_topic function for the corpus
+        all_topics = model.get_document_topics(self.corpus)
+
+        print(list(all_topics))
+
+        for topic in all_topics:
+            self.assertTrue(isinstance(topic, list))
+            for k, v in topic:  # list of doc_topics
+                self.assertTrue(isinstance(k, numbers.Integral))
+                self.assertTrue(np.issubdtype(v, float))
+
+        # FIXME: Fails on osx and win
```

Comment:
what's about now (same question about other FIXMEs)?

Reply:
I'll uncomment that and check if something fails

Reply:
`per_word_topics` option currently is not implemented, removed that piece for now.